application-admintools-ui/src/main/resources/AdminTools/Code/AdminToolsJS.xml

```diff
@@ -123,7 +123,89 @@
       <cache>default</cache>
     </property>
     <property>
-      <code>require(['jquery'], function($) {
+      <code>require(['jquery', 'xwiki-meta', 'xwiki-job-runner'], function($, xwikiMeta, JobRunner) {
+  const getResourceURL = function() {
+    let documentReference = XWiki.Model.resolve('AdminTools.Code.HealthCheckResources', XWiki.EntityType.DOCUMENT);
+    return new XWiki.Document(documentReference).getURL('get');
+  };
+
+  const runHealthCheckJob = function() {
+    var data = [];
+    data.push(
+      {name: 'outputSyntax', value: 'plain'},
+      {name: 'sheet', value: 'wiki:AdminTools.Code.HealthCheckResources'},
+      {name: 'action', value: 'healthCheck'},
+      {name: 'form_token', value: xwikiMeta.form_token}
+    );
+    return Promise.resolve(new JobRunner({
+      createStatusRequest: function(jobId) {
+        return {
+          url: getResourceURL(),
+          data: {
+            outputSyntax: 'plain',
+            sheet: 'AdminTools.Code.HealthCheckResources',
+            data: 'jobStatus',
+            jobId: jobId.join('/')
+          }
+        };
+      }
+    }).run(getResourceURL(), data)).catch((reason) => {
+      return Promise.reject(reason);
+    }).finally(() => {
+      $('#admin_tools_job_start').prop("disabled", false);
+    });
+  };
+
+  const reinitialiseProgressBar = function() {
+    $.get(getResourceURL(), function(data) {
+      let progressBar = $(data).find('div.ui-progress');
+      $('#healthCheck .successmessage').replaceWith(progressBar)
+      $('#healthCheck .warningmessage').replaceWith(progressBar)
+      $('#healthCheck .errormessage').replaceWith(progressBar)
+    });
+  };
+
+  // Need to call job.min.js only after the job has started its execution and the result message has been changed for
+  // the progress bar. Otherwise, job.js won't see .ui-progress and will not initialize and run a new JobRunner
+  // with the requested configuration. To make sure the job .ui-progress is seen, we need to undefine and request the
+  // resource again.
+  const initialiseJobJS = function() {
+    require.undef('health-check-job');
+    $.get(getResourceURL(), function(data) {
+      let jobMinJsURL = $(data).find('input[name="jobMinJsURL"]').val();
+      require.config({
+        paths: {
+          'health-check-job': jobMinJsURL
+        }
+      });
+      require(['health-check-job'], function(){});
+    });
+  };
```

Comment:
This looks very hackish. We need to find a better solution (I need to understand the problem first :) ).

Reply:
The issue is when I try to run the job. After a job runs, the progress bar is replaced with the result message ([job.js replace line](https://github.com/xwiki/xwiki-platform/blob/63714c025a1da068946a170148772c2227009c02/xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-war/src/main/webapp/resources/uicomponents/job/job.js#L210)). For job.js to properly initialize the JobRunner and to run, the job progress must exist: [job.js JobRunner initialization](https://github.com/xwiki/xwiki-platform/blob/63714c025a1da068946a170148772c2227009c02/xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-war/src/main/webapp/resources/uicomponents/job/job.js#L216-L257). If I request job.js from the initialization, it will not detect that the result message was replaced with the progress bar and no progress will be shown in the UI.

The page after an execution looks like this: ![image](https://github.com/xwikisas/application-admintools/assets/104432589/5c882ab3-ca94-4e59-944e-52cfc29d76cd)

Reply:
I understand. The need is basically to reload ``job.js`` so that it detects the job started dynamically (without a page reload). Then I have a few comments:

* You should report an issue on XWiki Platform, because ``job.js`` should ideally be able to detect jobs started without a page reload (e.g. using ``xwiki:dom:updated`` event).
* You shouldn't mention "min". Whether the JavaScript is loaded minified or not is determined by the XWiki build and the way you access the page (e.g. by adding ``?minify=false`` to the URL query string).
* You don't need to request again the URL for the job.js each time you run the health check. This should be done only once, preferably when the page loads. You can pass the URL as an attribute of the health check trigger button (along with other "static" information you need on the client).
* Same, you shouldn't need to redefine (configure) the ``job.js`` RequireJS module each time you trigger a health check. Can be done only once (e.g. on page load, when you get the ``job.js`` URL)
* This means that when a health check is triggered you should simply undefine the module and request it again.